### PR TITLE
Add k3s mod sync script + test coverage output

### PR DESCRIPTION
**Pull request by dereknola**
- Added brads [k3s-modsync.sh script](https://gist.githubusercontent.com/brandond/5346acc3108b489fa2208b4eab4a1738/raw/3a226e20d9478f4b8aac9c9ec840284299815ccf/k3s-modsync.sh) (converted from original's bash to sh). Makes sure that the "replace" modules we pin to upstream k8s are up to date. Mainly used for new minor releases.
- Added a list and markdown table output for the test coverage tool. Helpfuly in tracking progress in the GH epic. 
Signed-off-by: Derek Nola <derek.nola@suse.com>

### CODE REVIEW

**Comment by briandowns** [comment]
These will need to be updated to conform to the repo standards.

**Comment by dereknola** [comment]
Could you more specific on how to best conform to repo standards? Looking at the README I made changes so that the spellcheck posix checks pass, as well as running `golangci-lint` and `gofmt -w` on my test-coverage changes reveals no issues.

File: bin/k3s-modsync.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
```
Comment (briandowns) [inline bin/k3s-modsync.sh:1]: Can you make this 4 space indentation as well as rename the file to `k3s_modsync.sh`

**Review by briandowns — APPROVED** [review]
LGTM

File: bin/k3s_modsync.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+K8S_REPO="kubernetes/kubernetes"
+K8S_COMMIT="${K8S_COMMIT:-}"
+if [ -z "${K8S_COMMIT}" ]; then
```
Comment (johnatasr) [inline bin/k3s_modsync.sh:5]: Suggestion :safety_pin: .  Use [[ ... ]] for conditional tests: The use of double brackets [[ ... ]] is preferred over single brackets [ ... ] for conditional tests in modern shell scripts, as it provides more advanced features such as improved string handling and pattern matching.

File: cmd/test_coverage/coverage.go [hunk 3]
```diff
@@ -265,60 +265,114 @@ func coverage(c *cli.Context) error {
...
+func graphResults(serverFlagSet map[string]int, vagrantCoverage []TestCov, intCoverage []TestCov) {
+	data := grob.Traces{}
+	xFlagNames := []string{}
+
+	for k := range serverFlagSet {
+		xFlagNames = append(xFlagNames, k)
+	}
+	for _, test := range append(vagrantCoverage, intCoverage...) {
+		var testGroup string
+		if group := "integration"; strings.Contains(test.shortPath, group) {
```
Comment (johnatasr) [inline cmd/test_coverage/coverage.go:335]: Another suggestion, this code couldn't be simplified ? Example:

`var testGroup string
if strings.Contains(test.shortPath, "integration") {
    testGroup = "integration"
} else if strings.Contains(test.shortPath, "install") {
    testGroup = "install"
} else if strings.Contains(test.shortPath, "e2e") {
    testGroup = "e2e"
}`

File: bin/k3s_modsync.sh (hunk 2 above)
Comment (dereknola) [reply bin/k3s_modsync.sh:5]: We write POSIX shell scripts in this repo, not bash, so double brackets are not an option.

File: cmd/test_coverage/coverage.go (hunk 3 above)
Comment (dereknola) [reply cmd/test_coverage/coverage.go]: Good call

File: bin/k3s_modsync.sh (hunk 2 above)
Comment (johnatasr) [reply bin/k3s_modsync.sh:5]: Sure, thank you !

**Review by johnatasr — APPROVED** [review]

**Pull request merged**